### PR TITLE
Add an ingestor which fans log entries out to appenders

### PR DESCRIPTION
**Pull request by dinowernli**
@devnev

### CODE REVIEW

File: appender/appender.go [hunk 1]
```diff
@@ -5,14 +5,15 @@ import (
...
 )
 
+// Appender handles requets to append log entries to storage-backed chunks.
 type Appender struct {
 	Storage            storage.Storage
 	MaxEntriesPerChunk int
 	openChunk          storage.LogChunk
 }
 
 // HandleRequest processes a request to append a single entry to the logs.
```
Comment (devnev) [inline appender/appender.go]: update comment

File: ingester/ingester.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	Discovery *discovery.Discovery
+}
+
+// HandleIngest ingests a single log entry.
+func (i *Ingester) HandleIngest(entry string) error {
+	// For now, just append to all appenders.
+	for _, appender := range i.Discovery.Appenders {
+		err := appender.HandleAppend(entry)
+		if err != nil {
+			return fmt.Errorf("Unable to handle append: %v", err)
```
Comment (devnev) [inline ingester/ingester.go]: The error wrapping would be more useful if it gave info about the loop state, i.e. the index for now (some name for the appender in the future would be nice). Also I just remembered that https://github.com/golang/go/wiki/CodeReviewComments#error-strings suggests lowercase error strings for cleaner concatenation.

File: main.go [hunk 3]
```diff
@@ -19,12 +21,27 @@ func main() {
...
+		},
 	}
+	discovery := &discovery.Discovery{Appenders: appenders}
+	ingester := &ingester.Ingester{Discovery: discovery}
 
 	for i := 0; i < 40; i++ {
-		appender.HandleRequest(fmt.Sprintf("entry-%d", i))
+		err := ingester.HandleIngest(fmt.Sprintf("entry-%d", i))
+		if err != nil {
+			log.Panicf(fmt.Sprintf("Failed to handle ingest, %v", err))
```
Comment (devnev) [inline main.go]: Similar to the error above, the message here would be more helpful if it provided context like the loop variable

**Review by devnev — CHANGES_REQUESTED** [review]
LGTM

File: appender/appender.go (hunk 1 above)
Comment (dinowernli) [reply appender/appender.go]: Done,

File: ingester/ingester.go (hunk 2 above)
Comment (dinowernli) [reply ingester/ingester.go]: Good points. Lower cased and added loop index.

File: main.go (hunk 3 above)
Comment (dinowernli) [reply main.go]: Done.

**Review by devnev — APPROVED** [review]

**Comment by dinowernli** [comment]
PTAL

**Pull request merged**